lang/rust/Config.in

```diff
@@ -0,0 +1,121 @@
+# Rust Language Options
+menu "Compiler Options"
+	depends on PACKAGE_rust
+
+	config RUST_DEBUG
+		bool "Enables Debugging Environment (--enable-debug)"
+		default n
+
+	config RUST_DOCS
+		bool "Build standard library documentation (--enable-docs)"
+		default n
+
+	config RUST_COMPILER_DOCS
+		bool "Build compiler documentation (--enable-compiler-docs)"
+		default n
+
+	config RUST_OPTIMIZE_TESTS
+		bool "Build tests with optimizations (--enable-optimized-tests)"
+		default n
+
+	config RUST_PARALLEL
+		bool "Build with multi-threaded support (--enable-parallel-compiler)"
+		default n
+
+	config RUST_VERBOSE_TESTS
+		bool "Enable verbose output when running tests (--enable-verbose-tests)"
+		default n
+
+	config RUST_CCACHE
+		bool "Build with ccache enabled (--enable-ccache)"
+		default n
+
+	config RUST_LLVM_STATIC
+		bool "Statically link to libstdc++ to LLVM (--enable-llvm-static-stdccp)"
+		default n
+
+	config RUST_LLVM_SHARED
+		bool "Prefer shared linking to LLVM (--enable-llvm-link-shared)"
+		default n
+
+	config RUST_CODEGEN_TESTS
+		bool "Run the src/test/codegen tests (--enable-codegen-tests)"
+		default n
+
+	config RUST_OPTION_CHECKING
+		bool "Complain about unrecognized options in this configure script (--enable-option-checking)"
+		default y
+
+	config RUST_ENABLE_NINJA
+		bool "Build LLVM using the Ninja generator (--enable-ninja)"
+		default y
+
+	config RUST_LOCKED_DEPS
+		bool "Force Cargo.lock to be up to date (--enable-locked-deps)"
+		default n
+
+	config RUST_VENDOR
+		bool "Enable usage of vendored Rust crates (--enable-vendor)"
+		default n
+
+	config RUST_SANITIZERS
+		bool "Build the sanitizer runtimes (asan, lsan, msan, tsan) (--enable-sanitizers)"
+		default n
+
+	config RUST_DIST_SRC
+		bool "When building tarballs enables building a source tarball (--enable-dist-src)"
+		default n
+
+	config RUST_CARGO_NATIVE_STATIC
+		bool "Build static native libraries in Cargo (--enable-cargo-native-static)"
+		default n
+
+	config RUST_PROFILER
+		bool "Build the profiler runtime (--enable-profiler)"
+		default n
+
+	config RUST_FULL_TOOLS
+		bool "Build all tools (--enable-full-tools)"
+		default y
+
+	config RUST_MISSING_TOOLS
+		bool "Allow failures when building tools (--enable-missing-tools)"
+		default y
+
+	config RUST_USE_LIBCXX
+		bool "Build LLMV with libc++ (--enable-use-libcxx)"
+		default n
+
+	config RUST_CONTROL_FLOW_GUARD
+		bool "Enable Control Flow Guard (--enable-control-flow-guard)"
+		default n
+
+	config RUST_OPTIMIZE
+		bool "Build optimized rust code (--enable-optimize)"
+		default n
```

Comment:
Does this correspond to `optimize = false` in config.toml? Setting that to `false` isn't supported, turning it off by default seems like a bad idea.

https://github.com/rust-lang/rust/blob/9503ea19edbf01b9435e80e17d60ce1b88390116/config.toml.example#L332-L336

Reply:
There are only a few things I set in the TOML via `--set` having to do with the build system. (https://github.com/openwrt/packages/pull/13916/files#diff-000828b821201f172a6a3c1b97c5a89fd38753d46e7b0c11e475b7e61a99a24cR49)

These simply add (or not) the `--` flag indicated (`--enable-optimize` in this case) to the arguments passed to `./configure`

Reply:
Well right, but it seems strange to me to expose this flag at all when there are a bunch of others you've left out. A compiler with optimize=false is basically useless, it will take a half hour just to compiler the standard library.

Reply:
I copied the output from `./configure --help` to generate the list.  Which did I miss?  As we've discussed, I'm totally open to suggestions. :)

Reply:
Oh, I don't think you need to expose every option by hand, there's a whole ton of them:

<details>

```
> grep -E '^(#[^ ]|\[)' config.toml.example 
#profile = <none>
[llvm]
#download-ci-llvm = false
#skip-rebuild = false
#optimize = true
#thin-lto = false
#release-debuginfo = false
#assertions = false
#ccache = false
#ccache = "/path/to/ccache"
#version-check = true
#static-libstdcpp = false
#ninja = true
#targets = "AArch64;ARM;Hexagon;MSP430;Mips;NVPTX;PowerPC;RISCV;Sparc;SystemZ;WebAssembly;X86"
#experimental-targets = "AVR"
#link-jobs = 0
#link-shared = false
#version-suffix = "-rust-dev"
#clang-cl = '/path/to/clang-cl.exe'
#cflags = "-fextra-flag"
#cxxflags = "-fextra-flag"
#ldflags = "-Wl,extra-flag"
#use-libcxx = true
#use-linker = "lld"
#allow-old-toolchain = false
#polly = false
[build]
#doc-stage = 0
#build-stage = 1
#test-stage = 1
#dist-stage = 2
#install-stage = 2
#bench-stage = 2
#build = "x86_64-unknown-linux-gnu"
#host = ["x86_64-unknown-linux-gnu"]
#target = ["x86_64-unknown-linux-gnu"]
#build-dir = "build"
#cargo = "/path/to/bin/cargo"
#rustc = "/path/to/bin/rustc"
#rustfmt = "/path/to/bin/rustfmt"
#docs = true
#compiler-docs = false
#submodules = true
#fast-submodules = true
#gdb = "gdb"
#nodejs = "node"
#python = "python"
#locked-deps = false
#vendor = false
#full-bootstrap = false
#extended = false
#tools = ["cargo", "rls", "clippy", "rustfmt", "analysis", "src"]
#verbose = 0
#sanitizers = false
#profiler = false
#cargo-native-static = false
#low-priority = false
#configure-args = []
#local-rebuild = false
#print-step-timings = false
[install]
#prefix = "/usr/local"
#sysconfdir = "/etc"
#docdir = "share/doc/rust"
#bindir = "bin"
#libdir = "lib"
#mandir = "share/man"
#datadir = "share"
#infodir = "share/info"
#localstatedir = "/var/lib"
[rust]
#optimize = true
#debug = false
#download-rustc = false
#codegen-units = if incremental { 256 } else { 16 }
#codegen-units-std = 1
#debug-assertions = rust.debug (boolean)
#debug-assertions-std = rust.debug-assertions (boolean)
#debug-logging = rust.debug-assertions (boolean)
#debuginfo-level = 0
#debuginfo-level-rustc = 0
#debuginfo-level-std = 0
#debuginfo-level-tools = 0
#debuginfo-level-tests = 0
#run-dsymutil = false
#backtrace = true
#incremental = false
#parallel-compiler = false
#default-linker = "cc"
#channel = "dev"
#description = ""
#musl-root = "..."
#rpath = true
#verbose-tests = false
#optimize-tests = true
#codegen-tests = true
#ignore-git = true
#dist-src = false
#save-toolstates = "/path/to/toolstates.json"
#codegen-backends = ["llvm"]
#lld = false
#use-lld = false
#llvm-tools = false
#deny-warnings = true
#backtrace-on-ice = false
#verify-llvm-ir = false
#thin-lto-import-instr-limit = 100
#remap-debuginfo = false
#jemalloc = false
#test-compare-mode = false
#llvm-libunwind = 'no'
#control-flow-guard = false
#new-symbol-mangling = false
[target.x86_64-unknown-linux-gnu]
#cc = "cc"
#cxx = "c++"
#ar = "ar"
#ranlib = "ranlib"
#linker = "cc"
#llvm-config = "../path/to/llvm/root/bin/llvm-config"
#llvm-filecheck = "/path/to/FileCheck"
#android-ndk = "/path/to/ndk"
#sanitizers = false
#profiler = false
#crt-static = false
#musl-root = "..."
#musl-libdir = musl-root/lib
#wasi-root = "..."
#qemu-rootfs = "..."
[dist]
#sign-folder = "path/to/folder/to/sign"
#gpg-password-file = "path/to/gpg/password"
#upload-addr = "https://example.com/folder"
#src-tarball = true
#missing-tools = false
#compression-formats = ["gz", "xz"]
```

</details>

My point was that this option doesn't seem particularly useful to expose, and especially shouldn't be on by default.